maestro-orchestra/src/main/java/maestro/orchestra/Orchestra.kt

```diff
@@ -135,11 +139,73 @@ class Orchestra(
             is StopAppCommand -> maestro.stopApp(command.appId)
             is ClearStateCommand -> maestro.clearAppState(command.appId)
             is ClearKeychainCommand -> maestro.clearKeychain()
+            is RunFlowCommand -> runFlowCommand(command)
             is ApplyConfigurationCommand, null -> { /* no-op */
             }
         }
     }
 
+    private fun runFlowCommand(command: RunFlowCommand) {
+        if (evaluateCondition(command.condition)) {
+            runSubFlow(command.commands)
+        } else {
+            throw CommandSkipped
+        }
+    }
+
+    private fun evaluateCondition(condition: Condition?): Boolean {
+        if (condition == null) {
+            return true
+        }
+
+        condition.visible?.let {
+            try {
+                findElement(it, timeoutMs = optionalLookupTimeoutMs)
+            } catch (ignored: MaestroException.ElementNotFound) {
+                return false
+            }
+        }
+
+        condition.notVisible?.let {
+            val result = MaestroTimer.withTimeout(optionalLookupTimeoutMs) {
+                try {
+                    findElement(it, timeoutMs = 500L)
+
+                    // If we got to that point, the element is still visible.
+                    // Returning null to keep waiting.
+                    null
+                } catch (ignored: MaestroException.ElementNotFound) {
+                    // Element was not visible, as we expected
+                    true
+                }
+            }
+
+            // Element was actually visible
+            if (result != true) {
+                return false
+            }
+        }
+
+        return true
+    }
+
+    private fun runSubFlow(commands: List<MaestroCommand>) {
```

Comment:
This command is _almost_ like `executeCommands` but what makes it hard to merge the two is the fact that `onCommandFailed` callback can produce exception of its own, making it an awkward branching